src/common/components/matomoTracker/__tests__/MatomoTracker.test.js

```diff
@@ -0,0 +1,95 @@
+/* eslint-disable no-underscore-dangle */
+import { TRACK_TYPES } from '../constants';
+import MatomoTracker from '../MatomoTracker';
+
+const MOCK_URL_BASE = 'https://www.test.fi/';
+const MOCK_TRACKER_URL = 'https://www.test.fi/matomo.php';
+
+describe('MatomoTracker', () => {
+  beforeEach(() => {
+    window._paq = [];
+  });
+
+  it('should initialise window._paq', () => {
+    // eslint-disable-next-line no-new
+    new MatomoTracker({
+      urlBase: MOCK_URL_BASE,
+      siteId: 'test123',
+      srcUrl: 'test.js',
+      enabled: true,
+      configurations: {
+        foo: 'bar',
+        testArray: ['testArrayItem1', 'testArrayItem2'],
+        testNoValue: undefined,
+      },
+    });
+
+    expect(window._paq).toEqual([
+      ['setTrackerUrl', MOCK_TRACKER_URL],
+      ['setSiteId', 'test123'],
+      ['foo', 'bar'],
+      ['testArray', 'testArrayItem1', 'testArrayItem2'],
+      ['testNoValue'],
+      ['enableLinkTracking', true],
+    ]);
+  });
+
+  it('should throw error if urlBase missing', () => {
+    expect(() => new MatomoTracker({ siteId: 'test123' })).toThrowError();
+  });
+
+  it('should throw error if siteId missing', () => {
+    expect(
+      () =>
+        new MatomoTracker({
+          urlBase: 'http://www.test.fi',
```

Comment:
Sonar is complaining that this should be `https://`. Doesn't really matter but probably does not hurt to change it here 🙂